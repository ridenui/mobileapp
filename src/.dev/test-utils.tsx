// https://testing-library.com/docs/react-native-testing-library/setup
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { light } from '@styles/Themes';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={light}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from '@testing-library/react-native';
export * from 'jest-styled-components';

export { customRender as render };
