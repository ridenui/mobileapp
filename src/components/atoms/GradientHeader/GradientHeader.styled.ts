import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';

export const GradientHeader = styled(LinearGradient)`
  height: 250px;
  width: 130%;
  left: -15%;
  top: -10%;
  transform: rotate(-12deg);
`;
