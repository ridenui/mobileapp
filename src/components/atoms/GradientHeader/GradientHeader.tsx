import React from 'react';
import { Colors } from '@styles/Colors';
import * as S from './GradientHeader.styled';

/**
 * Colored Gradient from red-to-orange
 */
export function GradientHeader(): JSX.Element {
  const { red, orange } = Colors;
  return <S.GradientHeader colors={[red, orange]} start={{ x: 0, y: 0.5 }} end={{ x: 1, y: 0.5 }} />;
}
