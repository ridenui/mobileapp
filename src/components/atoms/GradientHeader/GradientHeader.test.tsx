import React from 'react';
import { render } from '@dev/test-utils';
import { GradientHeader } from './GradientHeader';

test('GradientHeader', () => {
  const { toJSON } = render(<GradientHeader />);
  expect(toJSON()).toMatchSnapshot();
});
