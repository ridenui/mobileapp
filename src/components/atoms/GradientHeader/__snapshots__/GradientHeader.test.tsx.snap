// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientHeader 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293011496,
      4294937647,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.5,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  style={
    Array [
      Object {
        "height": 250,
        "left": "-15%",
        "top": "-10%",
        "transform": Array [
          Object {
            "rotate": "-12deg",
          },
        ],
        "width": "130%",
      },
    ]
  }
/>
`;
